### PR TITLE
8330165: C2: make superword consistently use PhaseIdealLoop::register_new_node()

### PR DESCRIPTION
**Pull request by rwestrel**
Another set of changes from 8275202. There are cases in superword
where new nodes are not assigned control. I believe they are harmless
currently because superword is the last pass of optimizations. I also
cleaned up the code so it always uses `register_new_node()`. There are
a couple places where `intcon()` should be used.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8330165](https://bugs.openjdk.org/browse/JDK-8330165): C2: make superword consistently use PhaseIdealLoop::register_new_node() (**Enhancement** - P5)

### Reviewers
 * [Christian Hagedorn](https://openjdk.org/census#chagedorn) (@chhagedorn - **Reviewer**)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18760/head:pull/18760` \
`$ git checkout pull/18760`

Update a local copy of the PR: \
`$ git checkout pull/18760` \
`$ git pull https://git.openjdk.org/jdk.git pull/18760/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18760`

View PR using the GUI difftool: \
`$ git pr show -t 18760`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18760.diff">https://git.openjdk.org/jdk/pull/18760.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18760#issuecomment-2051671597)

### CODE REVIEW

File: src/hotspot/share/opto/superword.cpp [hunk 1]
```diff
@@ -2553,8 +2554,7 @@ bool SuperWord::output() {
         BasicType bt = velt_basic_type(cmp);
         const TypeVect* vt = TypeVect::make(bt, vlen);
         VectorNode* mask = new VectorMaskCmpNode(bol_test, cmp_in1, cmp_in2, bol_test_node, vt);
-        igvn().register_new_node_with_optimizer(mask);
-        phase()->set_ctrl(mask, phase()->get_ctrl(p->at(0)));
+        phase()->register_new_node(mask, phase()->get_ctrl(p->at(0)));
```
[the remark this answers is not in the document]
Comment (rwestrel) [reply src/hotspot/share/opto/superword.cpp]: Good idea. I updated the change.

**Review by chhagedorn — APPROVED** [review]
That looks good to me, thanks for additionally updating other uses of this pattern!

**Review by vnkozlov — APPROVED** [review]
Good.

**Comment by rwestrel** [comment]
@chhagedorn @vnkozlov thanks for the reviews

**Pull request closed without merging**